vespa/deployment.py

```diff
@@ -886,8 +1140,85 @@ def get_connection_response_with_retry(self, method, path, body, headers):
         self.connection.request(method, path, body, headers)
         return self.connection.getresponse()
 
+    def _try_get_access_token(self) -> str:
+        # Check if auth.json exists
+        if not self.auth_file_path.exists():
+            print("No auth.json found. Please authenticate.")
+            self._vespa_auth_login()
+            # Recheck for auth.json after authentication
+            if not self.auth_file_path.exists():
+                raise FileNotFoundError("Authentication failed, auth.json not found.")
+        else:
+            print("Checking for access token in auth.json...")
+
+        # Load the auth.json file
+        auth = json.loads(self.auth_file_path.read_text())
+
+        # Ensure the datetime string is parsed correctly
+        try:
+            expires_at = parser.parse(
+                auth["providers"]["auth0"]["systems"]["public"]["expires_at"]
+            )
+        except ValueError as e:
+            print(f"Error parsing the date: {e}")
+            raise
+
+        # Compare offset-aware datetime objects
+        if expires_at < datetime.now(timezone.utc):
+            print("Access token expired. Please re-authenticate.")
+            # Remove the expired file
+            os.remove(self.auth_file_path)
+            self._vespa_auth_login()
+            # Reload the auth.json file after re-authentication
+            auth = json.loads(self.auth_file_path.read_text())
+            try:
+                expires_at = parser.parse(
+                    auth["providers"]["auth0"]["systems"]["public"]["expires_at"]
+                )
+            except ValueError as e:
+                print(f"Error parsing the date: {e}")
+                raise
+            if expires_at < datetime.now(timezone.utc):
+                raise Exception("Authentication failed, token is still expired.")
+
+        return auth["providers"]["auth0"]["systems"]["public"]["access_token"]
+
+    def _request_access_token(
```

Comment:
Perhaps `_request_with_access_token`? Likewise for the one with `api_key` below.

Reply:
Yep, that is much more descriptive.